nixos/modules/hardware/openrazer.nix

```diff
@@ -0,0 +1,160 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.hardware.openrazer;
+
+  toPyBoolStr = b: if b then "True" else "False";
+
+  daemonExe = "${cfg.daemonPackage}/bin/openrazer-daemon --config ${daemonConfFile}";
+
+  daemonConfFile = pkgs.writeTextFile {
+    name = "razer.conf";
+    text = ''
+      [General]
+      verbose_logging = ${toPyBoolStr cfg.verboseLogging}
+
+      [Startup]
+      sync_effects_enabled = ${toPyBoolStr cfg.syncEffectsEnabled}
+      devices_off_on_screensaver = ${toPyBoolStr cfg.devicesOffOnScreensaver}
+      mouse_battery_notifier = ${toPyBoolStr cfg.mouseBatteryNotifier}
+
+      [Statistics]
+      key_statistics = ${toPyBoolStr cfg.keyStatistics}
+    '';
+  };
+
+  dbusServiceFile = pkgs.writeTextFile rec {
+    name = "org.razer.service";
+    destination = "/share/dbus-1/services/${name}";
+    text = ''
+      [D-BUS Service]
+      Name=org.razer
+      Exec=${daemonExe}
+      SystemdService=openrazer-daemon.service
+    '';
+  };
+
+  drivers = [
+    "razerkbd"
+    "razermouse"
+    "razerfirefly"
+    "razerkraken"
+    "razermug"
+    "razercore"
+  ];
+in
+{
+  options = {
+    hardware.openrazer = {
+      enable = mkEnableOption "OpenRazer drivers and userspace daemon.";
+
+      daemonPackage = mkOption {
+        type = types.package;
+        default = pkgs.python3Packages.openrazer-daemon;
+        defaultText = "pkgs.python3Packages.openrazer-daemon";
+        description = ''
+          Which package to use as the openrazer daemon.
+        '';
+      };
+
+      driverPackage = mkOption {
```

Comment:
I personally don't like such '*Package' options because chances are, nobody will ever need them, and overlays can still be used to change the used packages if absolutely necessary. Until somebody needs them, I think more options are just noise. People have differing opinions on this though, see https://github.com/NixOS/nixpkgs/issues/50476. In any case, if these options seem needed, they can always be added later.